azDevOps/azure/templates/v2/steps/test-functional-testcafe.yml

```diff
@@ -1,25 +1,25 @@
 ############################################################################################################
 # desc: Runs the functional (e2e) tests on a deployed webapp with Testcafe
-# params: Env vars, Node version, workingDirectory, build_number
-# return: Publishes test results, screenshots and video recordings
-# pre-reqs: dependency installation, build webapp
+# params: Env vars, workingDirectory, testcafe_browser_list (default to all)
+# return: Publishes test results, screenshots
+# pre-reqs: deployed webapp
 ############################################################################################################
 
 parameters:
-  env_vars: {}
-  workingDirectory: ''
```

Comment:
I don't actually see a problem with leaving it as env_vars and workingDirectory here since we are already in a specific context scope. 
so no need to re-specify - it's a bit like a class called testcafe to have a method that's called testcafeDoStuff.

at a caller level though this is correct 
`testcafe_e2e_env_vars:
                PORT: 3000`
that then passes this into the task as 
- step: testcafe
   parameters:
      env_vars: testcafe_e2e_env_vars

Reply:
Do we not need to iterate over them and assign them to `env` as they are used as env variables not run args?

Reply:
yeah but we are inside the step itself but until it gets there they are passed around as maps
```env:
      ${{ each var in parameters.env_vars }}:
        ${{ var.key }}: ${{ var.value }}```

Reply:
but anyway merging this in now